### DISCUSSION TITLE
Link multiple Vercel teams to single Github account?

### DISCUSSION THREAD

**Discussion by eivindml** [vercel/vercel#5266] [Help] [answered]
Hi, 

I'm a member of many Vercel teams as I'm a freelance developer, and I create a new team for each client, and add the as an owner. 

But all the code for each client usually lives in repos in my Github account. So I need to link each team to my single Github account. It looks like this isn't working, but it doesn't give me any errors either, so I'm not sure.

I've managed to link one team (let's call it "Team A") to my Github account, and everything works here. But if I try to add a new Github integration for a project in another team (or my private user account), it won't work. It displays what's shown in the picture below, and if I go through the process below, it still shows the "Install" button. Tried multiple times. And as soon as I delete the connection for "Team A", then I can add the integration again.

<img width="775" alt="Skjermbilde 2020-10-06 kl  13 12 12" src="https://user-images.githubusercontent.com/202450/95194447-92f6ac00-07d5-11eb-9e3e-88e6f5653641.png">

---

Also the "Import Project" button doesn't behave as expected. If I selected "Team B" in the dropdown, then go through "Import Project", it will still import the project to "Team A", since that's the only team that is connected to my Github account.

### Expected behavior
I would expect it to be possible to connect multiple teams to one GitHub account, or at the least show some error message when trying to connect a second team, so as to not confuse us.

## Question
Is this the behavior, or am I doing something wrong? 🙏🏻

**Comment by SokratisVidros** [comment]
Having the same issue. Any status updates on that? Another use case is when companies want to segregate production and development teams in Vercel.

**Reply by paulogdm** [reply]
We should have a possible solution in the upcoming weeks. And the connection will be "per-project" rather than "per-team".

**Reply by SokratisVidros** [reply]
Great , thanks!

**Reply by trompx** [reply]
@paulogdm Will it be possible to deploy a project by being a simple collaborator and not the owner of the repo? I host the client project on my github account in a private repository and set the client as collaborator but when I want to deploy it from the vercel client account, it asks to link the vercel account with my github account...

**Reply by paulogdm** [reply]
I don't have specific information for your use-case, but the plan is to make the integration ask for fewer permissions.

**Reply by trompx** [reply]
My goal is to let the client have their vercel account and pay for their hosting themself, while I manage code deployment. For some clients, I give them the source code, in that case I can just be a collaborator on their github repository, but sometimes I don't provide them the source code. In that case, I would need a way to push to a repository they don't have access to. I could have lived to build the code and push it to their repository before deployment but Vercel doesn't allow local build https://vercel.com/knowledge/deploying-locally-built-nextjs. Any idea how I could manage this?

**Comment by rodrigomf24** [comment]
@paulogdm do you have any updates regarding this topic? I am able to only link my github to a single vercel account, in my case I have my personal account and my work one, I want to able able to keep both vercel accounts but with authorization on the same github account. I don't need to use the github account for login, I just need it to be authorized on both vercel accounts.

**Reply by rodrigomf24** [reply]
Currently, if I link my github account to multiple vercel accounts,  it will keep it linked to the last login connection attempt.

**Reply by amyegan** [reply, marked as answer]
Hi @rodrigomf24! Vercel allows you to connect one Git account to a single Vercel account for login. For Projects, you can now connect to other GitHub accounts and orgs from a Vercel account or Team. You can learn more about it here: [Deploying a Git Repository](https://vercel.com/docs/concepts/git#deploying-a-git-repository)

**Reply by amyegan** [reply]
This discussion is a couple of years old now. Platform changes have been released since then and current issues seem to be slightly different from what was originally being asked.

If you're running into a similar problem, please [start a new discussion](https://github.com/vercel/community/discussions/new) with details about your situation 🙏

**Comment by AnnekaSillitoe** [comment]
Am having this exact issue now.

Have got a personal project on a personal account deployed via Vercel

My work also use Vercel and I am registered on their account, on their team via my work email... both these email address are linked to my single Github account however.

To deploy to one or the other i'm having to "relink" my single github account for each of them and remember to switch between each time as it boots me out of the other one.

Is there a solution for this that allows me to link my github account and a specific email address to each one only so that I may use that github account for both without having to do the song and dance of relinking each time?

**Reply by Balance8** [reply]
Same issue here

**Reply by maxtechera** [reply]
I am having the same issue, this was not happening to me a couple weeks ago. 
It used to allow me to signin with the email. I have both emails configured on my Github Account. 
I would expect Vercel deployments to allow me to deploy from any of the emails I have configured on the account.

**Reply by pugson** [reply]
same issue still. very annoying

**Reply by aheidelberg** [reply]
I also have this issue. One GitHub account for both personal and work projects but two Vercel accounts for those same projects. I'm only able to deploy a commit to the Vercel account which most recently setup to Sign in with GitHub. I also seem to have to use my personal GitHub email in the `user.email` git config field to be able to deploy even with the Vercel work account, which uses a different email when not signing in with GitHub.

**Reply by lommaj** [reply]
same issue - this is super annoying.  Your single github account can be linked to many different organizations but vercel access is managed by the internal email you get and its treating github login connections uniquely.

**Reply by lommaj** [reply]
I need to be able to link a single github account to multiple vercel login accounts.

**Comment by timtischler** [comment]
Bump!  Same issue here.

**Comment by adrivelasco** [comment]
same issue here, hope there will be a solution soon. 🙏 

My solution is disconnect and connect my github account to the vercel project I want to work on. Really annoying.

**Comment by y471n** [comment]
Same issue. Any updates?

**Comment by garrettboatman** [comment]
Bump. Seems like such a basic feature.
